packages/byods/src/data-source-service/index.ts

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable no-useless-catch */
+/* eslint-disable no-console */
+/* eslint-disable class-methods-use-this */
+/* eslint-disable import/no-extraneous-dependencies */
+
+import {v4 as uuidv4} from 'uuid';
+import {decodeJwt, JWTPayload} from 'jose';
+import DataSourceClient from '../data-source-client/index';
+
+export default class DataSourceService {
+  private dataSourceClient: DataSourceClient;
+  private timer: any;
+
+  /**
+   * Creates an instance of DataSourceService.
+   * @param {DataSourceClient} dataSourceClient - The  dataSourceClient instance to use for DataSourceService.
+   * @example
+   * const dataSourceService = new DataSourceService(dataSourceClient);
+   */
+
+  constructor(dataSourceClient: DataSourceClient) {
+    this.dataSourceClient = dataSourceClient;
+  }
+
+  /**
+   * This method refreshes the DataSource token using dataSourceId and tokenLifetimeMinute
+   * @param {string} dataSourceId The id of data source
+   * @param {number} tokenLifetimeMinutes The Life time minutes for the data source. <=1440
+   * @returns {Promise<void>}
+   * @example
+   * const dataSourceService = new DataSourceService(dataSourceClient); Create an instance of dataSourceService by passing instance of dataSourceClient.
+   * dataSourceService.refreshDataSourceToken(12345, 60); Invoke refreshDataSourceToken method by invoking dataSourceId and tokenLifetimeMinutes.
+   */
+
+  public async refreshDataSourceToken(
```

Comment:
can we move it to byods index.ts or datasourceclinet?

Reply:
Need to check with the team where we should move it either index.ts or dataSourceClient?

Reply:
**_Addressed!_**